### PR TITLE
[1.3] Version bump for protobuf, snappy-java

### PR DESCRIPTION
**Pull request by dbwiddis**
### Description

Bumps protobuf-java to 3.25.5 (Resolves CVE-2024-7254)
Bumps snappy-java to 1.1.10.7 (Resolves CVE-2023-43642)

### Check List
- ~[ ] Functionality includes testing.~
- ~[ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md), if applicable.~
- ~[ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose), if applicable.~

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by owaiskazi19** [comment]
You need to run `./gradlew updateSHAs` to update the jars for the dependencies.

**Comment by dbwiddis** [comment]
```
Could not determine the dependencies of task ':plugins:repository-azure:thirdPartyAudit'.
> Could not resolve all files for configuration ':plugins:repository-azure:runtimeClasspath'.
   > Could not find com.azure:azure-storage-blob:12.18.1.
     Searched in the following locations:
       - https://repo.maven.apache.org/maven2/com/azure/azure-storage-blob/12.18.1/azure-storage-blob-12.18.1.pom
```
But it's there: https://central.sonatype.com/artifact/com.azure/azure-storage-blob/versions

**Comment by dbwiddis** [comment]
I'll revert the azure-storage bump as it seems more complicated.

**Comment by dbwiddis** [comment]
OK, I think I got all the dependencies sorted out.  Azure needed a few other dependencies added, I just copied current versions from 2.x and everything seems to work.  Should be GTG once gradle check passes.

**Comment by dbwiddis** [comment]
Going to back out the commons-compress changes as well as they run afoul of tika.

I'm going to need help from other maintainers on the commons-compress and azure-storage blob CVEs 
CC: @andrross @reta

**Comment by reta** [comment]
> I'm going to need help from other maintainers on the commons-compress and azure-storage blob CVEs
> CC: @andrross @reta

Seems like you pulled it off @dbwiddis ? thanks!

**Comment by dbwiddis** [comment]
OK, 2 out of 5 bumps is better than 0.

File: plugins/repository-gcs/build.gradle [hunk 1]
```diff
@@ -61,7 +61,7 @@ dependencies {
   api 'com.google.api:api-common:1.8.1'
   api 'com.google.api:gax:1.54.0'
   api 'org.threeten:threetenbp:1.4.4'
-  api 'com.google.protobuf:protobuf-java-util:3.19.3'
+  api 'com.google.protobuf:protobuf-java-util:3.25.5'
```
Comment (owaiskazi19) [inline plugins/repository-gcs/build.gradle:64]:
```suggestion
  api 'com.google.protobuf:protobuf-java-util:${versions.protobuf}'
```

**Comment by dbwiddis** [comment]
> Seems like you pulled it off @dbwiddis ? thanks! **UPD**: Oh I see commons-compres is out, will surely help you out

I only managed to get protobuf and snappy-java.

`commons-compress` runs afoul of `tika` on ingest.
`azure-storage-blob` requires a lot more transitive dependencies and fails container tests that don't exist in 2.x.
`hadoop-minicluster` fails on some hdfs tests.

I don't know how many are solvable or if we should mark the CVEs as "not fixable due to backwards compatibility" but I hope others with more experience with them (and the 2.x bumps) can advise.

**Review by andrross — APPROVED** [review]

File: plugins/repository-gcs/build.gradle (hunk 1 above)
Comment (dbwiddis) [reply plugins/repository-gcs/build.gradle:64]: @owaiskazi19 I tried this yesterday but it failed:
```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':plugins:repository-gcs:updateShas'.
> Could not resolve all files for configuration ':plugins:repository-gcs:runtimeClasspath'.
   > Could not find com.google.protobuf:protobuf-java-util:${versions.protobuf}.
     Required by:
         project :plugins:repository-gcs
```
There's probably some other fix to make it work but it's not this simple.

**Pull request merged**